File: src/main/java/com/learning/leetcode/patterns/Squares.py
def square(n: int):
    for i in range(n):
        for j in range(n):
            print("* ", end="")
        print()

def hollowSquare(n: int):
    for i in range(n):
        for j in range(n):
            if i == 0 or j == 0 or i == n - 1 or j == n - 1:
                print(" * ", end="")
            else:
                print("   ", end="")
        print()


def squareArray(n: int):
    for i in range(2 * n - 1):
        for j in range(2 * n - 1):
            top = i
            left = j
            right = (2 * n - 1) - 1 - j
            bottom = (2 * n - 1) - 1 - i
            print(n - min(left, right, top, bottom), end=" ")
        print()


def main():
    print("square array")
    square(5)
    print("Hollow square array")
    hollowSquare(5)
    print("Matrix pattern")
    squareArray(4)


main()

File: src/main/java/com/learning/leetcode/patterns/Squares.java
package com.learning.leetcode.patterns;

import static java.lang.Math.min;

public class Squares {
    public static void main(String[] args) {
        int n = 5;
        System.out.println("Square pattern");
        square(n);
        System.out.println("Hollow Square pattern");
        hollowSquare(n);
        System.out.println("Matrix number pattern");
        matrixNumber(n);
    }

    private static void square(int n) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print(" * ");
            }
            System.out.println();
        }
    }

    private static void hollowSquare(int n) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i == 0 || i == n - 1 || j == 0 || j == n - 1) {
                    System.out.print(" * ");
                } else {
                    System.out.print("   ");
                }
            }
            System.out.println();
        }
    }

    private static void matrixNumber(int n) {
        for (int i = 0; i < 2 * n - 1; i++) {
            for (int j = 0; j < 2 * n - 1; j++) {
                int top = i;
                int left = j;
                int right = (2 * n - 1) - 1 - j;
                int bottom = (2 * n - 1) - 1 - i;
                int val = n - min(min(top, left), min(right, bottom));
                System.out.print(val + " ");
            }
            System.out.println();
        }
    }
}
